### PR TITLE
Invocation too vague

### PR DESCRIPTION
**Pull request by Eruanion**
## Related issue

Closes: #66 

## Description
The invocation expressions keys are too imprecise causing false positive matches against methods of the same name in other namespaces. This makes the key more specific and prevents this from happening. 

## Supplemental testing
Describe any testing done in addition to existing unit tests

## Additional context
Please provide any additional information related to this PR

---
*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

File: src/CTA.Rules.Analysis/RulesAnalysis.cs [hunk 1]
```diff
@@ -184,7 +184,7 @@ private bool AnalyzeChildren(FileActions fileAction, UstList<UstNode> children,
                         case IdConstants.InvocationIdName:
                             {
                                 InvocationExpression invocationExpression = (InvocationExpression)child;
-                                var compareToken = new InvocationExpressionToken() { Key = invocationExpression.MethodName, Namespace = invocationExpression.Reference.Namespace, Type = invocationExpression.SemanticClassType };
+                                var compareToken = new InvocationExpressionToken() { Key = invocationExpression.SemanticOriginalDefinition, Namespace = invocationExpression.Reference.Namespace, Type = invocationExpression.SemanticClassType };
                                 _rootNodes.Invocationexpressiontokens.TryGetValue(compareToken, out var token);
                                 if (token != null)
                                 {
```
Comment (jonlouie) [inline src/CTA.Rules.Analysis/RulesAnalysis.cs:190]: This looks like the right thing to do, but were any rules affected by this?

File: src/CTA.Rules.Update/ActionsRewriter.cs [hunk 2]
```diff
@@ -195,27 +195,35 @@ public override SyntaxNode VisitInvocationExpression(InvocationExpressionSyntax
             var symbols = _semanticModel.GetSymbolInfo(node);
             var newNode = (InvocationExpressionSyntax)base.VisitInvocationExpression(node);
 
-            foreach (var action in _fileActions.InvocationExpressionActions)
+            var symbol = symbols.Symbol;
+
+            if (symbol != null)
             {
```
Comment (jonlouie) [inline src/CTA.Rules.Update/ActionsRewriter.cs:201]: Optional: replace the `if` block with a guard statement to reduce nesting

Example: 
```csharp
if (someCondition)
{
    return newNode;
}

// do stuff down here

```

File: src/CTA.Rules.Analysis/RulesAnalysis.cs (hunk 1 above)
Comment (Eruanion) [reply src/CTA.Rules.Analysis/RulesAnalysis.cs:190]: Oh yep, I had to fix all the invocation expression rules. There were quite a few for Owin and a couple in the system.Web rules file. Unfortunately after doing some checking the only way to fix this problem was to make the rule from precise which necessitated changing all the rules unfortunately.

File: src/CTA.Rules.Update/ActionsRewriter.cs (hunk 2 above)
Comment (Eruanion) [reply src/CTA.Rules.Update/ActionsRewriter.cs]: Thanks, will follow this rule from now on.

File: src/CTA.Rules.Analysis/RulesAnalysis.cs (hunk 1 above)
Comment (jonlouie) [reply src/CTA.Rules.Analysis/RulesAnalysis.cs:190]: Got it, makes sense. If the unit tests are all still working then this should be ok

**Review by jonlouie — APPROVED** [review]

**Review by marknfawaz — APPROVED** [review]

**Pull request merged**